src/main.cpp

```diff
@@ -775,88 +775,110 @@ bool AcceptToMemoryPoolWorker(CTxMemPool &pool,
         /* Continuously rate-limit free (really, very-low-fee) transactions
          * This mitigates 'penny-flooding' -- sending thousands of free transactions just to
          * be annoying or make others' transactions take longer to confirm. */
-        // maximum feeCutoff in satoshi per byte
-        static const double maxFeeCutoff =
-            boost::lexical_cast<double>(GetArg("-maxlimitertxfee", DEFAULT_MAXLIMITERTXFEE));
-        // starting value for feeCutoff in satoshi per byte
-        static const double initFeeCutoff =
-            boost::lexical_cast<double>(GetArg("-minlimitertxfee", DEFAULT_MINLIMITERTXFEE));
+        // maximum nMinRelay in satoshi per byte
         static const int nLimitFreeRelay = GetArg("-limitfreerelay", DEFAULT_LIMITFREERELAY);
 
         // get current memory pool size
         uint64_t poolBytes = pool.GetTotalTxSize();
 
-        // Calculate feeCutoff in satoshis per byte:
-        //   When the feeCutoff is larger than the satoshiPerByte of the
+        // Calculate nMinRelay in satoshis per byte:
+        //   When the nMinRelay is larger than the satoshiPerByte of the
         //   current transaction then spam blocking will be in effect. However
         //   Some free transactions will still get through based on -limitfreerelay
-        static double feeCutoff = initFeeCutoff;
+        static double nMinRelay = dMinLimiterTxFee.value;
         static double nFreeLimit = nLimitFreeRelay;
         static int64_t nLastTime = GetTime();
         int64_t nNow = GetTime();
 
-        // When the mempool starts falling use an exponentially decaying ~24 hour window:
-        // nFreeLimit = nFreeLimit + ((double)(DEFAULT_LIMIT_FREE_RELAY - nFreeLimit) / pow(1.0 - 1.0/86400,
-        // (double)(nNow - nLastTime)));
-        nFreeLimit /= std::pow(1.0 - 1.0 / 86400, (double)(nNow - nLastTime));
+        static double _dMinLimiterTxFee = dMinLimiterTxFee.value;
```

Comment:
can these now just be nonstatic locals, defined inside the lock?

Reply:
it has to be static because we need it on each iteration and for each thread.  We only update it if the tweak changes.  We could use the tweak but then I want to avoid calling the tweak value for every txn for now since I think we have to look at adding a lock to the tweaks anyway.  I guess I'm prematurely avoiding having to take  the tweak lock.